apps/emqx_s3/src/emqx_s3_profile_http_pool_clients.erl

```diff
@@ -0,0 +1,35 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2022-2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_s3_profile_http_pool_clients).
```

Comment:
>Are you concerned about memory GC pressure of frequent map updates?
The map may be quite large, also I generally tend to avoid keeping data whose size depends on some external activity in the process state.